CMakeLists.txt

```diff
@@ -375,17 +374,53 @@ if(LINUX)
   add_definitions(-D__linux__)
 endif(LINUX)
 
-if(ENABLE_SHARED)
-  set(Boost_USE_STATIC_LIBS   OFF)
-else(ENABLE_SHARED)
-  set(Boost_USE_STATIC_LIBS   ON)
-endif(ENABLE_SHARED)
+# Boost
+option(WITH_SYSTEM_BOOST "require and build with system Boost" OFF)
+
+if (WITH_SYSTEM_BOOST)
+  if(ENABLE_SHARED)
+    set(Boost_USE_STATIC_LIBS OFF)
+  else()
+    set(Boost_USE_STATIC_LIBS ON)
+  endif()
+else()
+  set(BOOST_J 1 CACHE STRING "max jobs for Boost build")
```

Comment:
@tchaikov, @cbodley I think Sage made a ruling on the submodule approach, but I don't have a strong preference.  I think this works ok for now (external project)?  Casey mentioned where the objects are built--I think they could be moved into build, if needed.